packages/tpetra/core/test/ImportExport/Issue_3968.cpp

```diff
@@ -0,0 +1,206 @@
+//
+//          Tpetra: Templated Linear Algebra Services Package
+//                 Copyright (2008) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+*/
+
+#include "Teuchos_UnitTestHarness.hpp"
+
+#include "Tpetra_Core.hpp"
+#include "Tpetra_Map.hpp"
+#include "Tpetra_Vector.hpp"
+#include "Tpetra_Export.hpp"
+#include "Teuchos_CommHelpers.hpp"
+
+#include <iostream>
+#include <numeric>
+#include <vector>
+
+// Test for Trilinos GitHub Issue #3968.
+
+namespace { // (anonymous)
+
+  using GST = Tpetra::global_size_t;
+  using LO = Tpetra::Map<>::local_ordinal_type;
+
+  // Try to get a 64-bit GlobalOrdinal type; that should be more
+  // likely to manifest atomic update issues.  If we can't, though,
+  // then a 32-bit type is fine.  long long is guaranteed to be at
+  // least 64 bits, so prefer that over long or unsigned long.
+#if defined(HAVE_TPETRA_INST_INT_LONG_LONG)
+  using GO = long long; // always at least 64 bits
+#elif defined(HAVE_TPETRA_INST_INT_LONG)
+  using GO = long; // may be 64 or 32 bits
+#elif defined(HAVE_TPETRA_INST_INT_UNSIGNED_LONG)
+  using GO = unsigned long; // may be 64 or 32 bits
+#else
+  using GO = Tpetra::Map<>::global_ordinal_type;
+#endif
```

Comment:
Does this mean that the Tpetra default favors int over long long? I would have expected long long.

Reply:
@rppawlo Fun fact: Default GlobalOrdinal type is `int`, even though `long long` is enabled by default.  See https://github.com/trilinos/Trilinos/issues/4029 .